backend/tests/fightMessageParsing.test.ts

```diff
@@ -0,0 +1,23 @@
+// https://github.com/TotallyNotRobots/CloudBot/blob/992586301a567a27342f2f87240123821e0b821e/data/attacks/fight.json#L4
+
+import { IrcMessageProcessor } from "../irc/IrcMessageProcessor";
+
+describe('parsing a fight result', () => {
+  test('works for correct message (returns match with winner and loser) 1', async () => {
+    const msg = 'ZAM! BOOM! BANG! fear stands victorious over Asmodean with a crushing side kick.';
+    const result = IrcMessageProcessor.matchesFightMsg(msg);
+    expect(result).toStrictEqual({valid: true, winner: 'fear', loser: 'Asmodean'});
+  });
+
+  test('works for correct message (returns match with winner and loser) 2', async () => {
+    const msg = 'SLAM! BOOM! POW! someone is the champion over anyone with a fatal haymaker punch.';
+    const result = IrcMessageProcessor.matchesFightMsg(msg);
+    expect(result).toStrictEqual({valid: true, winner: 'someone', loser: 'anyone'});
+  });
```

Comment:
I had to update the regex slightly once already so that it matched an additional fight message. If it still doesn't catch all we can make it even less restrictive.